src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -4228,7 +4228,7 @@ bool GetWaterThermalTankInput(EnergyPlusData &state)
                     if (!(UtilityRoutines::SameString(HPWH.TankName, Tank.Name) && UtilityRoutines::SameString(HPWH.TankType, Tank.Type))) continue;
 
                     // save backup element and on/off-cycle parasitic properties for use during standard rating procedure
-                    HPWH.BackupElementCapacity = Tank.MaxCapacity;
+                    setBackupElementCapacity(HPWH.BackupElementCapacity, Tank.MaxCapacity);
```

Comment:
This is in GetWaterThermalTankInput. So if the tank max capacity is still equal to autosize here, which makes sense if sizing has not been called, then where is the backup element size set after GetWaterThermalTankInput is called? Is GetWaterThermalTankInput called more than once? If not, then there needs to be somewhere in init where the backup element is set using valid data? Since the comment mentions standard rating procedures, the backup element must be set before that call. Or another call to setBackupElementCapacity within the call to standard rating? since Tank.MaxCapacity would need to be known by then. Or right before the call to standard ratings inside somewhere that is a one time init.

Reply:
I see this same code in one other place. This same function used there as well?

    DesuperHtr.BackupElementCapacity = Tank.MaxCapacity;

Reply:
How about setting the backup element capacity right here in sizing?

                if (this->MaxCapacityWasAutoSized) {
                    this->MaxCapacity = tmpMaxCapacity;
                    BaseSizer::reportSizerOutput(state, this->Type, this->Name, "Maximum Heater Capacity [W]", this->MaxCapacity);
                }

Reply:
Setting the backup element capacity here probably means the new function is no longer needed.

Reply:
@rraustad I see what you are saying--setting the backup capacity in WaterThermalTankData::SizeStandAloneWaterHeater makes sense if it is autosized.  It would have to be done in several places in that routine since there are different sizing types for this unit.  I'll look into implementing this.

Reply:
But I still wonder if we don't want to set the backup capacity to -99999 just to be sure.  Maybe we don't need the new function.  In the case of a user inputed capacity, it can assign the backup capacity right away.  But if the backup capacity gets set to autosize, I feel like that's a bit dangerous potentially.  Maybe instead of a straight assignment, the code should say:

if (Tank.MaxCapacity < 0.0) HPWH.BackupElementCapacity = Tank.MaxCapacity;

If the BackupElementCapacity is initialized to zero, then it avoids the problematic situation where right now it gets set to -99999 and never recovers.
What do you think of that?  As in, get rid of the new function, modify the Get routine with the if from above and also for the Desuperheater, and then setting the backup capacity in the sizing routines (12 different places where this->MaxCapacity = tmpMaxCapacity happens).  Would that satisfy your concerns?  Biggest thing I haven't looked at yet is how to access the correct Backup Capacity.

Reply:
I think now that the new function should be for these new 12 assignment occurrences rather than how I have it set up now.  Since the Tank will need to reference a heat pump (this->HeatPumpNum) or a desuperheater (this->DesuperheaterNum), there would be good reason to not have to repeat this logic in all 12 places.  Then the unit test could be for this function not the one that I previously added.

Reply:
tmpMaxCapacity is set multiple times, MaxCapacity is only set once at the end of all those assignments. Set backup element capacity there and that should be it.

Reply:
There is this, if (this->MaxCapacityWasAutoSized), that lets us know that the MaxCapacity is autosized, there could also be one for backup element capacity but the backup is set equal to the MaxCapacity so it seems if MaxCapacity is autosized then they would still be equal after sizing. I am not positive on this but I did see several places where the max cap is reset for some reason. Worst case to me is if backup element capacity is autosized then set equal to MaxCapacity in sizing.

            // Reset the tank's internal heating element capacity.
            this->MaxCapacity = HeatPump.BackupElementCapacity;
            this->MinCapacity = HeatPump.BackupElementCapacity;

Reply:
You could also use:

    if (BackupElementCapacity == DataSizing::AutoSize) {
        BackupElementCapacity = MaxCapacity;
    }

which would be valid if this is set in getInput.

    HPWH.BackupElementCapacity = Tank.MaxCapacity;

Reply:
If I remember correctly (juggling 5 different defects in the review process right now so it's possible that I am wrong here), there are three different sizing routines and at least two different types of equipment (heat pump and desuperheater) that use different data structures.  I'm not sure it will be super easy, but I think I get your point.  It probably won't be as complex (maybe 3 different places rather than greater than 12) as I thought.  I will try to work on this soon (hopefully later today?) and get a new commit made.